llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -136,13 +140,95 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineVectorShiftsByZero(
   return true;
 }
 
+bool AIE2PreLegalizerCombinerImpl::tryToCombineSetExtract(
+    MachineInstr &MI) const {
+  const Register DstReg = MI.getOperand(0).getReg();
+  MachineInstr *ExtOp = MRI.getUniqueVRegDef(MI.getOperand(2).getReg());
```

Comment:
I think it is better to use getDefIgnoringCopies. We are in SSA here, so no need to worry about uniqueness.